google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/executor/AppEngineExecutor.java

```diff
@@ -42,11 +59,25 @@ public void cancel() {
     }
   }
 
-  private void setProcess(Process process) {
-    this.process = process;
-  }
-
-  public Process getProcess() {
-    return process;
+  /**
+   * Waits for the given process to finish, in a separate background thread, holding {@link
+   * CloudSdkServiceManager#getSdkReadLock()} lock.
+   */
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private void waitForProcessFinish(Process process) {
```

Comment:
this blocks the cloud SDK from running until this process is complete correct? Perhaps better naming to indicate this?

Reply:
like: `acquireCloudSdkReadLock`, or something along those lines?

Reply:
sounds good, maybe add something else to imply we wait for a process there as well? I.e. `acquireCloudSdkReadLockUntilFinished`?

Reply:
good point. how about `holdCloudSdkReadLock`? that should cover the whole concept I think